src/main/java/hudson/plugins/cppncss/CppNCSSPublisher.java

```diff
@@ -88,6 +91,9 @@ public String getDisplayName() {
         }
 
         public Publisher newInstance(StaplerRequest req, JSONObject formData) throws FormException {
+            if (req == null) {
+                throw new IllegalStateException("Request has not been passed to " + DescriptorImpl.class);
+            }
             ConvertUtils.register(CppNCSSHealthMetrics.CONVERTER, CppNCSSHealthMetrics.class);

```

Comment:
Better to just delete this override and do the `register` call in a `static` initializer.